docs/griptape-framework/engines/rag-engines.md

```diff
@@ -8,70 +8,101 @@ search:
 !!! note
     This section is a work in progress.
 
-`RagEngine` is an abstraction for implementing modular RAG pipelines.
-
-`RagContext` is a container object for passing around RAG context. 
+[Rag Engine](../../reference/griptape/engines/rag/index.md) is an abstraction for implementing modular retrieval augmented generation (RAG) pipelines.
 
 ### RAG Stages
-- `QueryRagStage` is for parsing and expanding queries.
-- `RetrievalRagStage` is for retrieving content.
-- `ResponseRagStage` is for augmenting and generating outputs.
+
+`RagEngine`s consist of three stages: `QueryRagStage`, `RetrievalRagStage`, and `ResponseRagStage`. Stages are always executed sequentially. `RagEngine`s are not meant to replace [Workflows](../structures/workflows.md); that's why they don't implement arbitrary DAGs.
```

Comment:
> Stages are always executed sequentially. `RagEngine`s are not meant to replace [Workflows]

Seems like the confusion would targeted more be towards Pipelines?

Reply:
I added this to address the workflows concern that already came up twice in other contexts. I'll mention pipelines as well.